addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -245,6 +245,7 @@ export class LinkPlugin extends Plugin {
             "[data-oe-model]",
             ":has(>[data-oe-model])",
         ],
+        keep_empty_link_predicates: (linkEl) => linkEl.hasAttribute("data-mimetype"),
```

Comment:
A predicate describes what something _is_, not what you're going to _do_, or _not do_ with it. Therefore it should be named something like `legit_empty_link_predicates` or `valid_empty_link_predicates`. Then what you do (or don't do) with legit/valid empty links is up to you.